jupyter_client/launcher.py

```diff
@@ -87,6 +93,10 @@ def launch_kernel(cmd, stdin=None, stdout=None, stderr=None, env=None,
             cwd = cast_bytes_py2(cwd, sys.getfilesystemencoding() or 'ascii')
             kwargs['cwd'] = cwd
 
+        # Windows store python will pass invalid paths for the connection file. 
+        # Turn the connection file into a real path
+        cmd = [ force_real_path(c) for c in cmd]
```

Comment:
Note that `cmd` represents Python code, not all elements of the list are paths. I don't think it is safe to assume that an element that is not a path won't exist as a path by chance.